atomic_reactor/build.py

```diff
@@ -183,6 +185,29 @@ def set_df_path(self, path):
         for image in dfp.parent_images:
             self.parent_images[image] = None
 
+        # validate user has not specified COPY --from=image
+        builders = []
+        for stmt in dfp.structure:
+            if stmt['instruction'] == 'FROM':
+                match = re.match(r'\S+\s+as\s+(\S+)', stmt['value'], re.I)
+                builders.append(match.group(1) if match else str(len(builders)))
+            if stmt['instruction'] == 'COPY':
+                match = re.match(r'--from=(\S+)', stmt['value'], re.I)
+                if not match:
+                    continue
+                stage = match.group(1)
+                if any(stage in [str(idx), builder] for idx, builder in enumerate(builders[:-1])):
```

Comment:
Probably worth a comment explaining why the last item in `builders` is ignored.

Reply:
Yeah. Actually... while it would be weird, I wonder if it would actually work for the stage to refer to itself there...

Reply:
Turns out self-reference *does* work and I could even imagine where it would be useful. So, removed the extra noise.